contracts/test/BrokenBlsAccount.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts/utils/Create2.sol";
+import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+
+import "../samples/SimpleAccount.sol";
+import "../bls/IBLSAccount.sol";
+
+/**
+ * for testing: a BLS account that fails to return its public-key (completely ignores its publickey)
+ * this is a copy of the normal bls account, but it returns a public-key unrelated to the one it is constructed with.
+ */
+contract BrokenBLSAccount is SimpleAccount, IBLSAccount {
+    address public immutable aggregator;
+
+    // The constructor is used only for the "implementation" and only sets immutable values.
+    // Mutable values slots for proxy accounts are set by the 'initialize' function.
+    constructor(IEntryPoint anEntryPoint, address anAggregator) SimpleAccount(anEntryPoint)  {
+        aggregator = anAggregator;
+    }
+
+    function initialize(uint256[4] memory aPublicKey) public virtual initializer {
+        (aPublicKey);
+        super._initialize(address(0));
+    }
+
+    function _validateSignature(UserOperation calldata userOp, bytes32 userOpHash, address userOpAggregator)
+    internal override view returns (uint256 sigTimeRange) {
+
+        (userOp, userOpHash);
+        require(userOpAggregator == aggregator, "BLSAccount: wrong aggregator");
+        return 0;
+    }
+
+    function getAggregator() external view returns (address) {
+        return aggregator;
+    }
+
+    function getBlsPublicKey() external override pure returns (uint256[4] memory) {
+        uint256[4] memory pubkey;
+        return pubkey;
+    }
+}
+
+
+/**
+ * Based n SimpleAccountFactory
+ * can't be a subclass, since both constructor and createAccount depend on the
+ * actual wallet contract constructor and initializer
+ */
+contract BrokenBLSAccountFactory {
```

Comment:
Maybe it is better to have a Base Factory so that we can change the implementation and `initialize` encoding without having to copy the entire contract? Now this is just for the tests, but in the future we may want different proxy account types?

I can make a PR if you think this is okay.

Reply:
I can hardly think you can extract a base factory. each method depends on actual wallet implementation and also constructor (and factory method) signatures differ - whether it is a single or multiple keys, or signature types, or recovery mechanism or whatever.